website/blog/2024-05-13-thallium.md

```diff
@@ -0,0 +1,326 @@
+---
+author: Tomasz Godzik
+title: Metals v1.3.1 - Thallium
+authorURL: https://twitter.com/<login>
+authorImageURL: https://github.com/<login>.png
+---
+
+We're happy to announce the release of Metals v1.3.1, which brings in a couple
+of useful improvements.
+
+<table>
+<tbody>
+  <tr>
+    <td>Commits since last release</td>
+    <td align="center">70</td>
+  </tr>
+  <tr>
+    <td>Merged PRs</td>
+    <td align="center">60</td>
+  </tr>
+    <tr>
+    <td>Contributors</td>
+    <td align="center">9</td>
+  </tr>
+  <tr>
+    <td>Closed issues</td>
+    <td align="center">27</td>
+  </tr>
+  <tr>
+    <td>New features</td>
+    <td align="center">3</td>
+  </tr>
+</tbody>
+</table>
+
+For full details:
+[https://github.com/scalameta/metals/milestone/67?closed=1](https://github.com/scalameta/metals/milestone/67?closed=1)
+
+Metals is a language server for Scala that works with VS Code, Vim, Emacs, Zed,
+Helix and Sublime Text. Metals is developed at the
+[Scala Center](https://scala.epfl.ch/) and [VirtusLab](https://virtuslab.com)
+with the help from contributors from the community.
+
+## TL;DR
+
+Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
+give Metals a try!
+
+- [Resolve classpath lazily in Bazel](#resolve-classpath-lazily-in-bazel)
+- [Replace `slowTask` with LSP progress](#replace-slowtask-with-lsp-progress)
+- [No workspace support](#no-workspace-support)
+
+## Resolve classpath lazily in Bazel
+
+Metals uses [Bazel BSP](https://github.com/jetBrains/bazel-bsp/) to interact if
+Bazel itself via the Build Server Protocol. Up until this release, we would
+eagerly resolve the classpath that is needed for some of the Metals features.
+However, this caused indexing times to be huge in large Bazel projects as it is
+expensive to calculate it. To avoid that we now only resolve classpath for
+targets users are currently working on.
+
+Let us know if you notice any more bottlenecks we can fix!
+
+## Replace `slowTask` with LSP progress
+
+There are several Language Server Protocol extensions that are used in Metals to
+provide features that go outside the LSP standard. This also included
+`Slow task` extension, but since a couple of LSP releases there has been a
+proper progress endpoint available. This is now used in Metals.
+
+This means that editors that did not implement the `slow task` extension will
+now properly show the progress of anything running within Metals. For anything
+that did implement it, there might be some visual changes. In Visual Studio Code
+the progress will be contained at the bottom in the status bar, which can be
+clicked on to see all current progress messages. Since, it will also contain all
+the current compilation, we removed `Ongoing compilations` section from the
+Metals view. This is possible thanks to the great work done by
+([kasiaMarek](https://github.com/kasiaMarek))
+
+## No workspace support
```

Comment:
Did you unrevert the change in VSCode?

Reply:
Will do before the release, thanks for reminding me!